test/functional/tests/conftest.py

```diff
@@ -161,36 +165,57 @@ def get_force_param(item):
     return item.config.getoption("--force-reinstall")
 
 
+def get_rpm_param(item):
+    return item.config.getoption("--rpm-install")
+
+
 def base_prepare(item):
     with TestRun.LOGGER.step("Cleanup before test"):
         TestRun.executor.run("pkill --signal=SIGKILL fsck")
         Udev.enable()
         kill_all_io()
         DeviceMapper.remove_all()
 
-        if installer.check_if_installed():
+        uninstall_cas = list(TestRun.item.iter_markers(name="uninstall_cas"))
+        installer = RpmInstaller if get_rpm_param(item) else Installer
+        opencas_installed = installer.check_if_installed()
+
+        if opencas_installed:
             try:
                 from api.cas.init_config import InitConfig
                 InitConfig.create_default_init_config()
                 unmount_cas_devices()
                 casadm.stop_all_caches()
                 casadm.remove_all_detached_cores()
             except Exception:
-                pass  # TODO: Reboot DUT if test is executed remotely
+                if TestRun.executor.is_remote():
+                    TestRun.executor.reboot()
 
         for disk in TestRun.dut.disks:
             disk.umount_all_partitions()
             disk.remove_partitions()
             create_partition_table(disk, PartitionTable.gpt)
 
-        if get_force_param(item) and not TestRun.usr.already_updated:
-            installer.rsync_opencas_sources()
+        # sources should be up-to-date on DUT even if installation is skipped
+        installer.rsync_opencas()
+
+        if uninstall_cas:
+            if opencas_installed:
+                RpmInstaller.uninstall_opencas()
+        elif get_force_param(item) and not TestRun.usr.already_updated:
             installer.reinstall_opencas()
```

Comment:
If RPM is installed and we want to reinstall from sources then installer.reinstall_opencas() will try to remove only source-based installation (and fail), and then will try to install from sources although RPM would still be installed.